### DISCUSSION TITLE
源码安装失败

### DISCUSSION THREAD

**Discussion by ttycc** [bfenetworks/bfe#1170] [General]
root@k8s-one:~/bfe# make
go: downloading github.com/baidu/go-lib v0.0.0-20200819072111-21df249f5e6a
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading gopkg.in/gcfg.v1 v1.2.3
go: downloading golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
go: downloading github.com/armon/go-radix v1.0.0
go: downloading github.com/abbot/go-http-auth v0.4.1-0.20181019201920-860ed7f246ff
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading gopkg.in/square/go-jose.v2 v2.4.1
go: downloading github.com/andybalholm/brotli v1.0.0
go: downloading github.com/miekg/dns v1.1.29
go: downloading golang.org/x/net v0.0.0-20211216030914-fe4d6282115f
go: downloading github.com/oschwald/geoip2-golang v1.4.0
go: downloading github.com/microcosm-cc/bluemonday v1.0.16
go: downloading github.com/russross/blackfriday/v2 v2.0.1
go: downloading go.uber.org/automaxprocs v1.4.0
go: downloading github.com/opentracing/opentracing-go v1.1.0
go: downloading go.elastic.co/apm v1.11.0
go: downloading github.com/uber/jaeger-client-go v2.22.1+incompatible
go: downloading github.com/openzipkin-contrib/zipkin-go-opentracing v0.4.5
go: downloading go.elastic.co/apm/module/apmot v1.7.2
go: downloading github.com/openzipkin/zipkin-go v0.2.2
go: downloading golang.org/x/sys v0.0.0-20211216021012-1d35b9e2eb4e
go: downloading github.com/uber/jaeger-lib v2.2.0+incompatible
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/zmap/go-iptree v0.0.0-20170831022036-1948b1097e25
go: downloading github.com/tjfoc/gmsm v1.3.2
go: downloading gopkg.in/warnings.v0 v0.1.2
go: downloading github.com/jehiah/go-strftime v0.0.0-20171201141054-1d33003b3869
go: downloading github.com/oschwald/maxminddb-golang v1.6.0
go: downloading github.com/aymerick/douceur v0.2.0
go: downloading github.com/shurcooL/sanitized_anchor_name v1.0.0
go: downloading github.com/opentracing-contrib/go-observer v0.0.0-20170622124052-a52f23424492
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/elastic/go-sysinfo v1.1.1
go: downloading go.elastic.co/fastjson v1.1.0
go: downloading go.elastic.co/apm/module/apmhttp v1.7.2
go: downloading go.uber.org/atomic v1.6.0
go: downloading github.com/asergeyev/nradix v0.0.0-20170505151046-3872ab85bb56
go: downloading github.com/modern-go/concurrent v0.0.0-20180228061459-e0a39a4cb421
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/gorilla/css v1.0.0
go: downloading google.golang.org/grpc v1.22.1
go: downloading golang.org/x/text v0.3.7
go: downloading howett.net/plist v0.0.0-20181124034731-591f970eefbb
go: downloading github.com/joeshaw/multierror v0.0.0-20140124173710-69b34d4ec901
go: downloading github.com/prometheus/procfs v0.0.3
installing goyacc
go install  golang.org/x/tools/cmd/goyacc
go: downloading golang.org/x/tools v0.1.8
goyacc installed
cd "bfe_basic/condition/parser" && go generate
go test -race -cover ./...
go: downloading github.com/stretchr/testify v1.7.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/davecgh/go-spew v1.1.1
?       github.com/bfenetworks/bfe      [no test files]
?       github.com/bfenetworks/bfe/bfe_basic    [no test files]
ok      github.com/bfenetworks/bfe/bfe_balance  0.300s  coverage: 64.3% of statements
ok      github.com/bfenetworks/bfe/bfe_balance/backend  0.179s  coverage: 60.5% of statements
ok      github.com/bfenetworks/bfe/bfe_balance/bal_gslb 0.421s  coverage: 32.8% of statements
ok      github.com/bfenetworks/bfe/bfe_balance/bal_slb  0.563s  coverage: 73.0% of statements
ok      github.com/bfenetworks/bfe/bfe_basic/action     0.380s  coverage: 77.5% of statements
ok      github.com/bfenetworks/bfe/bfe_basic/condition  0.156s  coverage: 46.8% of statements
ok      github.com/bfenetworks/bfe/bfe_basic/condition/parser   0.243s  coverage: 47.5% of statements
ok      github.com/bfenetworks/bfe/bfe_bufio    1.204s  coverage: 89.3% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_cluster_conf/cluster_conf     0.479s  coverage: 66.3% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_cluster_conf/cluster_table_conf       0.350s  coverage: 36.8% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_cluster_conf/gslb_conf        0.583s  coverage: 49.1% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_conf  0.731s  coverage: 70.5% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_route_conf/host_rule_conf     0.182s  coverage: 76.8% of statements
?       github.com/bfenetworks/bfe/bfe_config/bfe_tls_conf/server_cert_conf     [no test files]
?       github.com/bfenetworks/bfe/bfe_debug    [no test files]
ok      github.com/bfenetworks/bfe/bfe_config/bfe_route_conf/route_rule_conf    0.286s  coverage: 91.3% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_route_conf/vip_rule_conf      0.137s  coverage: 90.3% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_tls_conf/session_ticket_key_conf      0.385s  coverage: 56.2% of statements
ok      github.com/bfenetworks/bfe/bfe_config/bfe_tls_conf/tls_rule_conf        0.488s  coverage: 50.7% of statements
ok      github.com/bfenetworks/bfe/bfe_fcgi     0.180s  coverage: 66.4% of statements
?       github.com/bfenetworks/bfe/bfe_http/httputil    [no test files]
ok      github.com/bfenetworks/bfe/bfe_http     0.755s  coverage: 48.7% of statements
?       github.com/bfenetworks/bfe/bfe_modules  [no test files]
?       github.com/bfenetworks/bfe/bfe_modules/mod_logid        [no test files]
ok      github.com/bfenetworks/bfe/bfe_http2    4.932s  coverage: 80.0% of statements
ok      github.com/bfenetworks/bfe/bfe_http2/hpack      1.688s  coverage: 85.6% of statements
ok      github.com/bfenetworks/bfe/bfe_module   0.304s  coverage: 73.0% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_access       0.296s  coverage: 6.1% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_auth_basic   0.180s  coverage: 80.3% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_auth_jwt     0.036s  coverage: 81.8% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_auth_request 0.033s  coverage: 78.7% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_block        0.061s  coverage: 80.9% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_compress     0.082s  coverage: 69.1% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_cors 0.042s  coverage: 77.4% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_doh  0.027s  coverage: 65.3% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_errors       0.031s  coverage: 11.4% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_geo  0.063s  coverage: 78.8% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_header       0.051s  coverage: 56.9% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_http_code    0.035s  coverage: 64.3% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_key_log      0.035s  coverage: 39.4% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_markdown     0.051s  coverage: 85.9% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_prison       1.048s  coverage: 24.5% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_redirect     0.032s  coverage: 81.2% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_rewrite      0.034s  coverage: 58.9% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_secure_link  0.035s  coverage: 71.6% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_static       0.043s  coverage: 84.3% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_tag  0.032s  coverage: 85.6% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_tcp_keepalive        0.064s  coverage: 72.9% of statements
?       github.com/bfenetworks/bfe/bfe_modules/mod_trace/trace  [no test files]
?       github.com/bfenetworks/bfe/bfe_modules/mod_trace/trace/elastic  [no test files]
?       github.com/bfenetworks/bfe/bfe_modules/mod_trace/trace/jaeger   [no test files]
?       github.com/bfenetworks/bfe/bfe_modules/mod_trace/trace/zipkin   [no test files]
ok      github.com/bfenetworks/bfe/bfe_modules/mod_trace        0.042s  coverage: 70.6% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_trust_clientip       0.030s  coverage: 65.7% of statements
?       github.com/bfenetworks/bfe/bfe_route/bfe_cluster        [no test files]
ok      github.com/bfenetworks/bfe/bfe_modules/mod_userid       1.025s  coverage: 73.3% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_waf  0.153s  coverage: 84.4% of statements
ok      github.com/bfenetworks/bfe/bfe_modules/mod_waf/waf_rule 0.275s  coverage: 78.6% of statements
ok      github.com/bfenetworks/bfe/bfe_net/textproto    0.539s  coverage: 70.8% of statements
ok      github.com/bfenetworks/bfe/bfe_proxy    0.639s  coverage: 90.2% of statements
ok      github.com/bfenetworks/bfe/bfe_route    0.142s  coverage: 10.6% of statements
ok      github.com/bfenetworks/bfe/bfe_route/trie       0.653s  coverage: 96.4% of statements
ok      github.com/bfenetworks/bfe/bfe_server   0.043s  coverage: 0.1% of statements
?       github.com/bfenetworks/bfe/bfe_util/string_reverse      [no test files]
ok      github.com/bfenetworks/bfe/bfe_spdy     4.534s  coverage: 78.1% of statements
ok      github.com/bfenetworks/bfe/bfe_stream   4.874s  coverage: 89.0% of statements
        github.com/bfenetworks/bfe/bfe_tls      coverage: 34.6% of statements
panic: test timed out after 10m0s
running tests:
        TestHandshakeClientCertECDSA (10m0s)

goroutine 49 [running]:
testing.(*M).startAlarm.func1()
        /usr/local/lib/go/src/testing/testing.go:2241 +0x219
created by time.goFunc
        /usr/local/lib/go/src/time/sleep.go:176 +0x48

goroutine 1 [chan receive]:
testing.(*T).Run(0xc00013e340, {0x822bf9, 0x1c}, 0x849210)
        /usr/local/lib/go/src/testing/testing.go:1630 +0x82e
testing.runTests.func1(0x0?)
        /usr/local/lib/go/src/testing/testing.go:2036 +0x8e
testing.tRunner(0xc00013e340, 0xc00010fb40)
        /usr/local/lib/go/src/testing/testing.go:1576 +0x217
testing.runTests(0xc00011c460?, {0xa481a0, 0x2e, 0x2e}, {0x1c?, 0x1e?, 0xa4c3e0?})
        /usr/local/lib/go/src/testing/testing.go:2034 +0x87d
testing.(*M).Run(0xc00011c460)
        /usr/local/lib/go/src/testing/testing.go:1906 +0xb45
main.main()
        _testmain.go:192 +0x2fd

goroutine 47 [select]:
net.(*pipe).read(0xc0001aed00, {0xc0003c74de, 0x1, 0x0?})
        /usr/local/lib/go/src/net/pipe.go:159 +0x24f
net.(*pipe).Read(0x0?, {0xc0003c74de, 0x1, 0x1})
        /usr/local/lib/go/src/net/pipe.go:142 +0x55
io.ReadAtLeast({0x7f9fca972118, 0xc0001aed00}, {0xc0003c7180, 0x35f, 0x35f}, 0x35f)
        /usr/local/lib/go/src/io/io.go:332 +0xde
io.ReadFull(...)
        /usr/local/lib/go/src/io/io.go:351
github.com/bfenetworks/bfe/bfe_tls.(*clientTest).run(0xc00016f718, 0xc00019a9c0, 0x0)
        /root/bfe/bfe_tls/handshake_client_test.go:221 +0xb05
github.com/bfenetworks/bfe/bfe_tls.runClientTestForVersion(0xc00019c9d8?, 0xc00016fd70, {0x81b5aa, 0x7}, {0x81add0, 0x5})
        /root/bfe/bfe_tls/handshake_client_test.go:262 +0x288
github.com/bfenetworks/bfe/bfe_tls.runClientTestTLS10(...)
        /root/bfe/bfe_tls/handshake_client_test.go:266
github.com/bfenetworks/bfe/bfe_tls.TestHandshakeClientCertECDSA(0x0?)
        /root/bfe/bfe_tls/handshake_client_test.go:422 +0x519
testing.tRunner(0xc00019a9c0, 0x849210)
        /usr/local/lib/go/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run
        /usr/local/lib/go/src/testing/testing.go:1629 +0x806

goroutine 48 [select]:
net.(*pipe).read(0xc0001aec80, {0xc000311400, 0x400, 0x6f1ba0?})
        /usr/local/lib/go/src/net/pipe.go:159 +0x24f
net.(*pipe).Read(0x0?, {0xc000311400, 0x400, 0x400})
        /usr/local/lib/go/src/net/pipe.go:142 +0x55
github.com/bfenetworks/bfe/bfe_tls.(*block).readFromUntil(0xc000113890, 0xc0001b1c00, 0x5)
        /root/bfe/bfe_tls/conn.go:538 +0x2b7
github.com/bfenetworks/bfe/bfe_tls.(*Conn).readRecord(0xc0001b1c00, 0x14)
        /root/bfe/bfe_tls/conn.go:760 +0x5d9
github.com/bfenetworks/bfe/bfe_tls.(*clientHandshakeState).readFinished(0xc00016bcd0)
        /root/bfe/bfe_tls/handshake_client.go:519 +0xb3
github.com/bfenetworks/bfe/bfe_tls.(*Conn).clientHandshake(0xc0001b1c00)
        /root/bfe/bfe_tls/handshake_client.go:211 +0x27b9
github.com/bfenetworks/bfe/bfe_tls.(*Conn).Handshake(0xc0001b1c00)
        /root/bfe/bfe_tls/conn.go:1248 +0x1b7
github.com/bfenetworks/bfe/bfe_tls.(*Conn).Write(0xc0001b1c00, {0xc00016bf7a, 0x6, 0x6})
        /root/bfe/bfe_tls/conn.go:1116 +0xba
github.com/bfenetworks/bfe/bfe_tls.(*clientTest).run.func1()
        /root/bfe/bfe_tls/handshake_client_test.go:202 +0x8b
created by github.com/bfenetworks/bfe/bfe_tls.(*clientTest).run
        /root/bfe/bfe_tls/handshake_client_test.go:201 +0x4bb
FAIL    github.com/bfenetworks/bfe/bfe_tls      600.536s
ok      github.com/bfenetworks/bfe/bfe_util     0.589s  coverage: 23.7% of statements
ok      github.com/bfenetworks/bfe/bfe_util/access_log  0.720s  coverage: 22.6% of statements
ok      github.com/bfenetworks/bfe/bfe_util/bns 0.831s  coverage: 66.7% of statements
ok      github.com/bfenetworks/bfe/bfe_util/byte_pool   0.947s  coverage: 100.0% of statements
ok      github.com/bfenetworks/bfe/bfe_util/hash_set    0.158s  coverage: 90.6% of statements
ok      github.com/bfenetworks/bfe/bfe_util/ipdict      0.273s  coverage: 87.6% of statements
ok      github.com/bfenetworks/bfe/bfe_util/ipdict/txt_load     0.323s  coverage: 82.9% of statements
ok      github.com/bfenetworks/bfe/bfe_util/json        0.380s  coverage: 85.7% of statements
ok      github.com/bfenetworks/bfe/bfe_util/net_util    0.126s  coverage: 34.5% of statements
ok      github.com/bfenetworks/bfe/bfe_util/pipe        0.126s  coverage: 66.3% of statements
ok      github.com/bfenetworks/bfe/bfe_util/semver      0.239s  coverage: 72.7% of statements
ok      github.com/bfenetworks/bfe/bfe_util/signal_table        0.302s  coverage: 31.2% of statements
ok      github.com/bfenetworks/bfe/bfe_websocket        4.699s  coverage: 84.3% of statements
FAIL
make: *** [Makefile:102: test-case] Error 1
root@k8s-one:~/bfe# go version
go version go1.20 linux/amd64
root@k8s-one:~/bfe# lsb_release -a
LSB Version:    core-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.5 LTS
Release:        20.04
Codename:       focal

**Comment by cc14514** [comment]
- Currently, the highest supported version of Go for BFE is Go 1.9.x. 
- Please try compiling using Go versions 1.6.x to 1.9.x.
